src/chrome/local-runtime.ts

```diff
@@ -356,33 +355,34 @@ export default class LocalRuntime {
   }
 
   // Returns the S3 url or local file path
-  async returnScreenshot(): Promise<string> {
-    const data = await screenshot(this.client)
-
-    // check if S3 configured
-    if (
-      process.env['CHROMELESS_S3_BUCKET_NAME'] &&
-      process.env['CHROMELESS_S3_BUCKET_URL']
-    ) {
-      const s3Path = `${cuid()}.png`
-      const s3 = new AWS.S3()
-      await s3
-        .putObject({
-          Bucket: process.env['CHROMELESS_S3_BUCKET_NAME'],
-          Key: s3Path,
-          ContentType: 'image/png',
-          ACL: 'public-read',
-          Body: new Buffer(data, 'base64'),
-        })
-        .promise()
-
-      return `https://${process.env['CHROMELESS_S3_BUCKET_URL']}/${s3Path}`
-    } else {
-      // write to `${os.tmpdir()}` instead
-      const filePath = path.join(os.tmpdir(), `${cuid()}.png`)
-      fs.writeFileSync(filePath, Buffer.from(data, 'base64'))
+  async returnScreenshot(
+    selector?: string,
+    options?: ScreenshotOptions,
+  ): Promise<string> {
+    if (selector) {
+      if (this.chromelessOptions.implicitWait) {
+        this.log(`screenshot(): Waiting for ${selector}`)
+        await waitForNode(
+          this.client,
+          selector,
+          this.chromelessOptions.waitTimeout,
+        )
+      }
 
-      return filePath
+      const exists = await nodeExists(this.client, selector)
+      if (!exists) {
+        throw new Error(
+          `screenshot(): node for selector ${selector} doesn't exist`,
+        )
+      }
+    }
+
+    const data = await screenshot(this.client, selector)
+
+    if (isS3Configured()) {
+      return await uploadToS3(data, 'image/png')
+    } else {
+      return writeToFile(data, 'png', options && options.filePath)
```

Comment:
Is `filePath` truly required? Otherwise this might pass an object to `writeToFile`